packages/plugin-ext/src/plugin/plugin-manager.ts

```diff
@@ -97,25 +97,44 @@ export class PluginManagerExtImpl implements PluginManagerExt, PluginManager {
         this.messageRegistryProxy = this.rpc.getProxy(PLUGIN_RPC_CONTEXT.MESSAGE_REGISTRY_MAIN);
     }
 
-    $stopPlugin(contextPath: string): PromiseLike<void> {
-        this.activatedPlugins.forEach(plugin => {
-            if (plugin.stopFn) {
-                plugin.stopFn();
-            }
+    async $stop(pluginId?: string): Promise<void> {
+        if (!pluginId) {
+            this.stopAll();
+            return;
+        }
+        // TODO what to do about transitive extensions?
```

Comment:
I don't think we need this TODO, as transitive plugins are installed as proper first class plugins, so they will be discarded as well. Keeping it might distract readers.